Content.Server/_CD/Engraving/EngraveableSystem.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Content.Server.Administration;
+using Content.Server.Administration.Logs;
+using Content.Server.Popups;
+using Content.Shared.Database;
+using Content.Shared.Popups;
+using Content.Shared.Examine;
+using Content.Shared.Verbs;
+using Robust.Shared.Player;
+using Robust.Shared.Utility;
+
+namespace Content.Server._CD.Engraving;
+
+public sealed class EngraveableSystem : EntitySystem
+{
+    [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly QuickDialogSystem _dialog = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<EngraveableComponent, ExaminedEvent>(OnExamined);
+        SubscribeLocalEvent<EngraveableComponent, GetVerbsEvent<ActivationVerb>>(AddEngraveVerb);
+    }
+
+    private void OnExamined(Entity<EngraveableComponent> ent, ref ExaminedEvent args)
+    {
+        var msg = new FormattedMessage();
+        msg.AddMarkupOrThrow(Loc.GetString(ent.Comp.EngravedMessage == string.Empty
+            ? ent.Comp.NoEngravingText
+            : ent.Comp.HasEngravingText));
+
+        if (ent.Comp.EngravedMessage != string.Empty)
+            msg.AddMarkupOrThrow(Loc.GetString(ent.Comp.EngravedMessage));
```

Comment:
engraving ] will throw

Reply:
could probably use `AddMarkupPermissive` but seems to work fine either way? dialog might be sanitizing it already in TryParseQuickDialog or something
![image](https://github.com/user-attachments/assets/b59b36e4-e78c-4915-b46e-699dae3ea515)

Reply:
eh i'd leave it as is personally, if someone wants to do colors, it's nice to have
plenty of descriptions use colors anyways

Reply:
Changed this to ``AddMarkupPermissive`` as Milon suggested!